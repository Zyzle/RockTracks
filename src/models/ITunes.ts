export interface Result {
  artistId: number;
  collectionId: number;
  trackId: number
  artistName: string;
  collectionName: string;
  trackName: string;
  trackViewUrl: string;
  artworkUrl30: string;
  artworkUrl100: string;
  trackPrice: number;
  releaseDate: string;
  trackTimeMillis: number;
  currency: string;
}

export interface SearchResults {
  resultsCount: number;
  results: Result[]
}