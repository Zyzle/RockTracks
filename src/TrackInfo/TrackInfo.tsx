import React, { PureComponent } from 'react';
import { RouteComponentProps } from 'react-router-dom';

import { Duration, DateTime } from 'luxon';

import { Result } from '../models/ITunes';

import './TrackInfo.scss';

type TrackInfoProps =  Result & RouteComponentProps;

class TrackInfo extends PureComponent<TrackInfoProps> {

  constructor(props: TrackInfoProps) {
    super(props);
  }

  parsedTrackTime() {
    return Duration.fromMillis(this.props.trackTimeMillis).toFormat('m:ss');
  }

  parsedReleaseDate() {
    return DateTime.fromISO(this.props.releaseDate).toLocaleString(DateTime.DATE_SHORT);
  }

  render() {
    return (
      <div>
        <img src={this.props.artworkUrl100} alt={this.props.trackName} />
        <h1>{this.props.trackName}</h1>
        <h2>{this.props.artistName}</h2>
        <p>Price: ${this.props.trackPrice}</p>
        <p>Runtime: {this.parsedTrackTime()} - Release Date: { this.parsedReleaseDate() }</p>
        <a target="_blank" href={this.props.trackViewUrl}><button>more details</button></a>
      </div>
    );
  }
}

export default TrackInfo;