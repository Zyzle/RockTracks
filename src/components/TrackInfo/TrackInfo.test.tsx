import React from 'react';
import { RouteComponentProps } from 'react-router';

import renderer from 'react-test-renderer';;

import TrackInfo from './TrackInfo';
import { Result } from '../../models/ITunes';

describe('TrackInfo Component', () => {

  it('should render', () => {
    const props: Result & RouteComponentProps = {
      artistId: 123,
      collectionId: 456,
      trackId: 789,
      artistName: 'An artist',
      collectionName: 'A collection',
      trackName: 'A track',
      trackViewUrl: 'https://www.google.com',
      artworkUrl100: 'https://via.placeholder.com/100',
      artworkUrl30: 'https://via.placeholder.com/30',
      trackPrice: 1.99,
      releaseDate: '01-01-1970',
      trackTimeMillis: 80000,
      currency: 'GBP',
      history: jest.fn() as any,
      location: jest.fn() as any,
      match: jest.fn() as any
    };

    const component = renderer.create(
      <TrackInfo  {...props} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});