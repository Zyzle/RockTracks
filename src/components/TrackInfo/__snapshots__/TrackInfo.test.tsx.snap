// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackInfo Component should render 1`] = `
<div>
  <img
    alt="A track"
    src="https://via.placeholder.com/100"
  />
  <h1>
    A track
  </h1>
  <h2>
    An artist
  </h2>
  <p>
    Price: $
    1.99
  </p>
  <p>
    Runtime: 
    1:20
     - Release Date: 
    Invalid DateTime
  </p>
  <a
    href="https://www.google.com"
    target="_blank"
  >
    <button>
      more details
    </button>
  </a>
</div>
`;
